site/content/articles/external-container-registry-integration.adoc

```diff
@@ -0,0 +1,479 @@
+...
+status:
+  internalRegistryHostname: image-registry.openshift-image-registry.svc:5000
+----
+
+More information can be found in the link:https://docs.openshift.com/container-platform/4.2/openshift_images/image-configuration.html#images-configuration-insecure_image-configuration[Image Configuration documentation]
+
+==== Insecure Registry With OpenShift v3
+
+To configure the docker daemon to trust content from an insecure registry, add the following to the *OPTIONS* property in the `/etc/sysconfig/docker` file. An example is shown below:
+
+[source]
+----
+OPTIONS=' --selinux-enabled --insecure-registry=external-registry.example.com:5000
+----
+
+NOTE: The hostname or IP address and optional port value must match the remote registry value configured in OpenShift
+
+Restart the docker daemon for the new configurations to take effect
+
+[source]
+----
+systemctl restart docker
+----
+
+== Accessing Secure Registries
+
+Secure external registries can be integrated into the OpenShift ecosystem. Through the use of secrets, users can provide the platform with the necessary credentials to access secured resources.
+
+=== Storing Credentials in Secrets
+
+Docker stores the details for authenticating to remote repositories in one of two formats: a config.json (.dockerconfigjson) file or a legacy .dockercfg file. OpenShift makes use of the .dockercfg file format for authenticating against remote registries and stores them as secrets, but for the purpose of this guide, only the .dockerconfigjson file format will be illustrrated. This file can be generated by using the `docker login` command or the parameters supplied to OpenShift who in turn will produce a new .dockerconfigjson file in the secret.
+
+Existing config.json files can be added as secrets by executing the following command:
+
+[source]
+----
+oc create secret external-registry --from-file=.dockerconfigjson=<path/to/.docker/config.json> --type=kubernetes.io/dockerconfigjson
+----
+
+Alternatively, a `.dockerconfigjson` formatted file can be created by supplying the necessary parameters using the following command:
+
+[source]
+----
+oc create secret docker-registry external-registry \
+    --docker-username=mastermind \
+    --docker-password=12345 \
+    --docker-server=external-registry.example.com:5000
+----
+
+NOTE: When logging into a remote OpenShift cluster, You must use your oc login token instead of your user's password. You can see more here. link:https://docs.openshift.com/container-platform/4.2/registry/accessing-the-registry.html[Accessing The Registry Directly]
+
+
+== External registry integration in Builds
+
+An external registry can be integrated into the OpenShift build process as both a location to store images built in the platform and as an image source for builds.
+
+
+=== Storing images as a result of a build
+
+NOTE: It is recommended that images produced as a result of a build within OpenShift be stored within the image registry integrated into the platform.
```

Comment:
Can we provide some information as to WHY we recommend this? It looks like we are basically opening this section with "You shouldn't do this, but here are the instructions anyway." It would be better if we could provide some example scenarios or use cases where this would be acceptable, but make it clear that this is an exception, not the rule.